src/caffe/layers/l1_loss_layer.cpp

```diff
@@ -0,0 +1,59 @@
+#include "caffe/layers/l1_loss_layer.hpp"
+
+#include <vector>
+
+#include "caffe/layer.hpp"
+#include "caffe/util/io.hpp"
+#include "caffe/util/math_functions.hpp"
+
+namespace caffe {
+
+template <typename Dtype>
+void L1LossLayer<Dtype>::Reshape(
+  const vector<Blob<Dtype>*>& bottom, const vector<Blob<Dtype>*>& top) {
+  LossLayer<Dtype>::Reshape(bottom, top);
+  CHECK_EQ(bottom[0]->count(1), bottom[1]->count(1))
+      << "Inputs must have the same dimension.";
+  diff_.ReshapeLike(*bottom[0]);
+  sign_.ReshapeLike(*bottom[0]);
+}
+
+
+template <typename Dtype>
+void L1LossLayer<Dtype>::Forward_cpu(const vector<Blob<Dtype>*>& bottom,
+                                            const vector<Blob<Dtype>*>& top) {
+  int count = bottom[0]->count();
+  caffe_sub(
+      count,
+      bottom[0]->cpu_data(),
+      bottom[1]->cpu_data(),
+      diff_.mutable_cpu_data());
+  caffe_cpu_sign(count, diff_.cpu_data(), sign_.mutable_cpu_data());
+  Dtype abs_sum = caffe_cpu_asum(count, diff_.cpu_data());
+  Dtype loss = abs_sum / bottom[0]->num();
```

Comment:
I think you should divide by `count` here to normalize the loss. By the way `num()` is deprecated - [see](http://caffe.berkeleyvision.org/doxygen/classcaffe_1_1Blob.html#a56c2b25db397d9e82bbd7c43597ae427)